doc/programming/parsing.rst

```diff
@@ -187,6 +187,40 @@ There's also a programmatic way to change a field's type to something
 that's different than what's being parsed, see the
 :ref:`attribute_convert`.
 
+.. _attribute_size:
+
+.. rubric:: Limiting Input Size
+
+You can limit the input that a field receives for parsing by attaching
+a ``&size=EXPR`` attribute that specifies the number of bytes to make
+available:
+
+.. spicy-code:: size.spicy
+
+    module Test;
+
+    public type Foo = unit {
+        x: bytes &until=b"X" &size=5;
+        y: bytes &eod;
+        on %done { print self; }
+    };
+
+.. spicy-output:: size.spicy
+    :exec: printf '12345678X90' | spicy-driver %INPUT
+    :show-with: foo.spicy
+
+As you can see, parsing `x` stops at the size limit, not at the
+delimiter coming only later.
```

Comment:
Is the logic here to consume the minimum number of bytes to eiter see `&until`'s arg or `&size` bytes? If it is I think it would be worth spelling that out.

I see one can also attach `&size` to e.g., an `uint8` field. If our explanation here could explain what happens in that case as well the even better.

Reply:
Yeah, that's what it means. Thinking about it, though, I think the semantics are still not quite right. The example in the doc should actually trigger a parse error because it's not finding the `X` before the imposed size limit (that's how `until` normal works when encountering EOD). I'll give that another try.